### PR TITLE
Use caching to speed up CICD

### PR DESCRIPTION
**Pull request by zjaco13**
*Issue #, if available:*

*Description of changes:*
Creates a new ecr repo for caching and uses it during build process

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf
Comment (5herlocked) [inline platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf:1]: Unless this is required, drop files that aren't required for your changes. It is a regression as I see it.

File: platform/backstage/templates/cicd-pipeline/provisioner/wf-templates.yaml
Comment (5herlocked) [inline platform/backstage/templates/cicd-pipeline/provisioner/wf-templates.yaml:1]: This seems to do everything that @shapirov103's CI/CD system (already merged) does?

**Comment by 5herlocked** [comment]
Hey seems like there's a bunch of misc changes in here that repeat work? Please double check and let's merge it into the files that already exist.

File: platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf [hunk 1]
```diff
@@ -45,7 +45,7 @@ module "eks" {
     static_ng = {
       use_custom_launch_template = false
       launch_template_name       = ""
-      instance_types             = ["m5.large"]
+      instance_types             = ["m5.2xlarge"]
 
       min_size     = 2
```
Comment (shapirov103) [inline platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf:50]: let's add desired_size = 2

File: platform/backstage/templates/cicd-pipeline/provisioner/wf-templates.yaml
Comment (shapirov103) [reply platform/backstage/templates/cicd-pipeline/provisioner/wf-templates.yaml:1]: This PR is to apply caching to images and speed up the builds. Zach authored the original pipeline, my contribution was fairly modest.

File: platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf
Comment (shapirov103) [reply platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf:1]: The change to the eks setup was discussed: we really don't want to wait for karpenter to spin up a box every time we submit a build job.

File: platform/backstage/templates/cicd-pipeline/template-cicd-pipeline.yaml [hunk 2]
```diff
@@ -70,6 +70,20 @@ spec:
           aws_region: ${{ parameters.aws_region }}
           hostname: ${{parameters.hostname}}
           gitea_token: ${{secrets.gitea_token}}
+          dockerfile_path: ${{parameters.dockerfile_path}}
+
+    #- action: fetch:template
```
Comment (shapirov103) [inline platform/backstage/templates/cicd-pipeline/template-cicd-pipeline.yaml]: do we need this commented section? If not, let's drop

**Review by shapirov103 — COMMENTED** [review]
Some minor comments, please take a look. 

Was it tested? Just trying to figure out if there may be any regression. 

Will the warmer workflow be run at provisioning time and if yes, will it run docker build as part of it?

**Comment by zjaco13** [comment]
> Some minor comments, please take a look.
> 
> Was it tested? Just trying to figure out if there may be any regression.
> 
> Will the warmer workflow be run at provisioning time and if yes, will it run docker build as part of it?

Yes, the warmer workflow runs as part of a workflow-of-workflows after the provisioner workflow.  The warmer workflow runs a full docker build to populate the cache in ECR(names the image cache-init).  I did test the full workflow successfully with dotnet(build down to 1.5 min), and tested the two workflows separately with rust(build down to 3 min).

File: platform/backstage/templates/cicd-pipeline/template-cicd-pipeline.yaml (hunk 2 above)
Comment (zjaco13) [reply platform/backstage/templates/cicd-pipeline/template-cicd-pipeline.yaml]: done

File: platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf (hunk 1 above)
Comment (zjaco13) [reply platform/infra/terraform/mgmt/terraform/mgmt-cluster/eks.tf:50]: I believe thats already specified, line 52

**Review by shapirov103 — APPROVED** [review]
LGTM - great work, will test out shortly

**Pull request merged**